examples/mpi/run_batched_mpi_async.py

```diff
@@ -0,0 +1,84 @@
+    server_agent_config.server_configs.scheduler_kwargs.num_clients = num_clients
+    if hasattr(server_agent_config.server_configs.aggregator_kwargs, "num_clients"):
+        server_agent_config.server_configs.aggregator_kwargs.num_clients = num_clients
+    # Create the server agent and communicator
+    server_agent = ServerAgent(server_agent_config=server_agent_config)
+    server_communicator = MPIServerCommunicator(comm, server_agent, logger=server_agent.logger)
+    # Start the server to serve the clients
+    server_communicator.serve()
+else:
+    # Create client agents for each client in the batch
+    client_agents: List[ClientAgent] = []
+    client_agent_config = OmegaConf.load(args.client_config)
+    for client_id in client_batch[rank - 1]:
+        client_agent_config.train_configs.logging_id = f'Client{client_id}'
+        client_agent_config.data_configs.dataset_kwargs.num_clients = num_clients
+        client_agent_config.data_configs.dataset_kwargs.client_id = client_id
+        client_agent_config.data_configs.dataset_kwargs.visualization = True if client_id == 0 else False
+        client_agents.append(ClientAgent(client_agent_config=client_agent_config))
+    # Create the client communicator for batched clients
+    client_communicator = MPIClientCommunicator(comm, server_rank=0, client_ids=client_batch[rank - 1])
+    # Get and load the general client configurations
+    client_config = client_communicator.get_configuration()
+    for client_agent in client_agents:
+        client_agent.load_config(client_config)
+    # Get and load the initial global model
+    init_global_model = client_communicator.get_global_model(init_model=True)
+    for client_agent in client_agents:
+        client_agent.load_parameters(init_global_model)
+    # Send the sample size to the server
+    client_sample_sizes = {
+        client_id: {
+            'sample_size': client_agent.get_sample_size(),
+            'sync': True
+        } 
+        for client_id, client_agent in zip(client_batch[rank - 1], client_agents)
+    }
+    client_communicator.invoke_custom_action(action='set_sample_size', kwargs=client_sample_sizes)
+    
+    # Generate data readiness report
+    if hasattr(client_config.data_readiness_configs, 'generate_dr_report') and client_config.data_readiness_configs.generate_dr_report:
+        data_readiness = {
+            client_id: client_agent.generate_readiness_report(client_config)
+            for client_id, client_agent in zip(client_batch[rank - 1], client_agents)
+        }
+        client_communicator.invoke_custom_action(action='get_data_readiness_report', kwargs=data_readiness)
+    
+    # Local training and global model update iterations
+    finish_flag = False
+    while True:
+        for client_id, client_agent in zip(client_batch[rank - 1], client_agents):
+            client_agent.train()
+            client_local_model = client_agent.get_parameters()
```

Comment:
Do we want to consider the case when `client_local_model` is a tuple here?

Reply:
Hi @yli697, yes, we need to consider that, I have fixed this.